File: front/amadeusSports/src/app/dto/MatchDto.ts
import { LeagueDto } from "./LeagueDto";
import { TeamDto } from "./TeamDto";

export class MatchDto {
    id!: number;
    time!: string;
    team1!: TeamDto;
    team2!: TeamDto;
    sport!: string;
    score!: string;
    league!: LeagueDto;
    details!: string;
    isFavorite!: boolean;

    constructor(id: number, time: string, team1: TeamDto, team2: TeamDto, score:string, sport: string, league: LeagueDto,
        details: string, isFavorite:boolean) {
        this.time = time;
        this.details = details;
        this.league = league;
        this.sport = sport;
        this.team1 = team1;
        this.team2 = team2;
        this.score = score;
        this.isFavorite = isFavorite;
    }
}

File: back/src/main/java/com/example/back/controllers/dto/MatchDto.java
package com.example.back.controllers.dto;

import com.example.back.models.entities.Team;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class MatchDto {
    Long id;
    String time;
    TeamDto team1;
    TeamDto team2;
    String sport;
    String score;
    String country;
    String details;
    Boolean isFavorite;
    LeagueDto league;
    String winner;
}
